privval/remote_signer.go

```diff
@@ -160,6 +166,10 @@ func RegisterRemoteSignerMsg(cdc *amino.Codec) {
 	cdc.RegisterConcrete(&SignedProposalResponse{}, "tendermint/remotesigner/SignedProposalResponse", nil)
 	cdc.RegisterConcrete(&PingRequest{}, "tendermint/remotesigner/PingRequest", nil)
 	cdc.RegisterConcrete(&PingResponse{}, "tendermint/remotesigner/PingResponse", nil)
+	// TODO: Evaluate whether we need to make this an official remotesigner
+	// message. Right now it's only relevant to KMS, and is only used in
+	// testing.
```

Comment:
The idea was that tendermint can shut down the kms via this message. I can not think of a realistic scenario where this is useful though 🤔  

CC'ing @zmanian and @tarcieri who introduced this message on the kms AFAIR. 

If we add this to tendermint, we should also document this in the spec (maybe you did that and I haven't seen this yet; reviewing each commit separately).

Reply:
You're right - the only scenario that I'm catering for here is being able to programmatically (and gracefully) shut the KMS down once the harness is finished testing it.

An alternative approach here is to rather send a `SIGTERM` from outside the process during the testing over here: https://github.com/thanethomson/kms/blob/issues/154/tests/support/run-harness-tests.sh

It doesn't appear as though the KMS shuts down gracefully at present when I hit Ctrl+Break. Is this something related to the [Abscissa](https://github.com/iqlusioninc/abscissa/) framework KMS uses, or is this something we need to build into the KMS? It looks like it's something I'd have to build into the KMS (I'll handle that if necessary).

Reply:
Would be great @tarcieri, if there's a relatively easy to way to handle signals generically - alternatively I could just use a library like [`signal_hook`](https://docs.rs/signal-hook/0.1.7/signal_hook/) internally in KMS for now and refactor the [start command logic](https://github.com/tendermint/kms/blob/master/src/commands/start.rs#L52).

Reply:
Sounds good to me.